### PR TITLE
Enable debug logging of signature verification

### PR DESCRIPTION
**Pull request by DRvanR**
This is a request from @pmeulen so that debugging is easier when Response signature verfication fails.

- [ ] Inject the logger rather than pass in method call.

### CODE REVIEW

**Comment by DRvanR** [comment]
*poke* @rjkip

File: build-pre-commit.xml [hunk 1]
```diff
@@ -63,7 +63,6 @@
                 <tokenfilter>
                     <containsregex pattern="src/(?!Tests).*$"/>
                 </tokenfilter>
-                <deletecharacters chars="\n"/>
```
Comment (rjkip) [inline build-pre-commit.xml]: Aaaah, now what? ;(

Comment (DRvanR) [inline build-pre-commit.xml]: That broke the pre-commit hook :cry:

File: src/Signing/SignatureVerifier.php [hunk 2]
```diff
@@ -47,12 +55,20 @@ public function __construct(KeyLoader $keyLoader)
...
         });
 
+        $this->logger->debug(sprintf(
+            'Found "%d" X509 keys, attempting to use each for signature verification',
+            $x509Keys->count()
+        ));
+
         foreach ($x509Keys as $key) {
             if ($this->isSignedWith($request, $key)) {
                 return true;
```
Comment (rjkip) [inline src/Signing/SignatureVerifier.php]: I'd say a debug message informing the signature was (not) properly signed with one of the keys is useful and completes the log messages. The consumer can still log its process on INFO or whatever (we're throwing if signature validation fails) at its discretion.

Comment (DRvanR) [inline src/Signing/SignatureVerifier.php]: That is actually done in the `isSignedWith` method. If all keys fail, false is returned and that immediately triggers an exception. Not really sure if it adds anything to duplicate the `Signature VERIFIED` below here again, will add a statement that the signature could not be verified with any key.

**Comment by DRvanR** [comment]
Added log message, anything else?

File: src/Signing/SignatureVerifier.php (hunk 2 above)
Comment (rjkip) [inline src/Signing/SignatureVerifier.php]: Where? I can find only one usage of `hasValidSignature`, and that is in the [redirect binding](https://github.com/SURFnet/Stepup-saml-bundle/blob/0b43a05614c7229e142893eaedf343ad037e9020/src/Http/RedirectBinding.php#L123).

**Comment by rjkip** [comment]
To me it's weird that a signature verification, if successful, ends with the message 'attempting to use each for verification', instead of 'yay it workstest'

File: src/Signing/SignatureVerifier.php (hunk 2 above)
Comment (rjkip) [inline src/Signing/SignatureVerifier.php]: Ah yes, hasValidSignature calls it :sweat_smile:. My point still stands though.

**Comment by DRvanR** [comment]
See [this line](https://github.com/SURFnet/Stepup-saml-bundle/pull/15/files#diff-4d4cda5f7b6c1327c9d83667a3d2e9a1R91) for the statement about attempting verification of the signature with key X, and [this line](https://github.com/SURFnet/Stepup-saml-bundle/pull/15/files#diff-4d4cda5f7b6c1327c9d83667a3d2e9a1R96) that notifies that the signature was verified (3 lines lower it logs when it could not verify). So you get:

- log about extracting keys
- log how many keys found, filtering X509
- log how many X509 keys found, attempting to verify
- for each key:
    - log attempting to verify with key X, detailing key contents
    - if success: log Signature Verified => _exit with return true_
    - if fail: log signature not verified
- endforeach: log signature could not be verified with any of the keys => _exit with return false_

Duplicating that the signature was verified right after the message that the signature was verified which is right after the message detailing the key with contents is redundant imho.

**Comment by DRvanR** [comment]
As an aside: the requester was already very happy with what was logged :wink:

**Comment by rjkip** [comment]
I have yet to wake up properly. I agree with you. :+1: Ship it.

**Pull request merged**